packages/editor/src/editor/sidebar/fields/ColumnsCheckboxField.tsx

```diff
@@ -1,30 +1,37 @@
-import type { DataTable, DataTableColumn, Variable } from '@axonivy/form-editor-protocol';
-import { BasicCheckbox, Fieldset, Message, type BrowserNode } from '@axonivy/ui-components';
-import { useMeta } from '../../../context/useMeta';
-import { useAppContext } from '../../../context/AppContext';
-import { createId, useData } from '../../../data/data';
-import { findAttributesOfType } from '../../../data/variable-tree-data';
+import type { DataTableColumn } from '@axonivy/form-editor-protocol';
+import { BasicCheckbox, Flex, Message } from '@axonivy/ui-components';
+import { createId } from '../../../data/data';
+import { isSameColumn, useDataTableColumns } from '../../../data/useDataTableColumns';
+import { useEffect, useState } from 'react';
 
 type ColumnsCheckboxFieldProps = {
-  label: string;
-  columns: DataTableColumn[];
   onChange: (value: DataTableColumn[]) => void;
 };
 
-type CheckboxColumn = DataTableColumn & { selected: boolean };
+export type CheckboxColumn = DataTableColumn & { selected: boolean };
 
-export const ColumnsCheckboxField = ({ label, columns, onChange }: ColumnsCheckboxFieldProps) => {
-  const { context } = useAppContext();
-  const { element } = useData();
+export const ColumnsCheckboxField = ({ onChange }: ColumnsCheckboxFieldProps) => {
+  const { dataTableColumns, upToDateColumns, boundDataTableColumns, unboundDataTableColumns } = useDataTableColumns();
+  const [localUnbindedDataTableColumns, setLocalUnbindedDataTableColumns] = useState<CheckboxColumn[]>(unboundDataTableColumns);
```

Comment:
why such long names? the context is clear that we have here a data table...
also the whole handling about this unbound columns can be moved to the custom hook if this helps...